akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -257,7 +257,12 @@ private[io] class SelectionHandler(settings: SelectionHandlerSettings) extends A
                               decision: SupervisorStrategy.Directive): Unit =
         try {
           val logMessage = cause match {
-            case e: ActorInitializationException if e.getCause ne null ⇒ e.getCause.getMessage
+            case e: ActorInitializationException if (e.getCause ne null) && (e.getCause.getMessage ne null) ⇒ e.getCause.getMessage
+            case e: ActorInitializationException if (e.getCause ne null) ⇒
+              e.getCause match {
```

Comment:
You raise a good point and you seem to have experience with this topic, so help is definitely welcome. It would be best to first open tickets for discussing shortcomings and goals before diving into wide-sweeping change mode. It would be good to also include test cases that verify the improved logging so we don’t accidentally break it again later—would you mind adding one to this PR as well? Thanks!